tests/testthat/test-equals_na_linter.R

```diff
@@ -56,3 +56,7 @@ test_that("equals_na_linter blocks disallowed usages in edge cases", {
   # correct line number for multiline code
   expect_lint("x ==\nNA", list(line_number = 1L, column_number = 1L, ranges = list(c(1L, 4L))), linter)
 })
+
+test_that("x %in% NA is caught", {
+  expect_lint("x %in% NA", "Use is.na for comparisons to NA", equals_na_linter())
```

Comment:
Why not add this to the existing patrick case?

Reply:
Mainly b/c i was too preoccupied thinking about the 3 things mentioned in the description :)